src/main/kotlin/com/hj/leetcode/kotlin/problem2130/Solution.kt

```diff
@@ -0,0 +1,50 @@
+package com.hj.leetcode.kotlin.problem2130
+
+import com.hj.leetcode.kotlin.common.model.ListNode
+
+/**
+ * LeetCode page: [2130. Maximum Twin Sum of a Linked List](https://leetcode.com/problems/maximum-twin-sum-of-a-linked-list/);
+ */
+class Solution {
+    /* Complexity:
+     * Time O(N) and Space O(N) where N is the size of head;
+     */
+    fun pairSum(head: ListNode?): Int {
+        // Collect all values in the linkedList then compute all twin sums and return the maximum
+        return maxTwinSum(head.values())
+    }
+
+    private fun ListNode?.values(): List<Int> {
+        val result = mutableListOf<Int>()
+        linearTraversal { node -> result.add(node.`val`) }
+        return result
+    }
+
+    private fun ListNode?.linearTraversal(onEachNode: (node: ListNode) -> Unit) {
+        var currentNode = this
+
+        while (currentNode != null) {
+            onEachNode(currentNode)
+            currentNode = currentNode.next
+        }
+    }
+
+    private fun maxTwinSum(values: List<Int>): Int {
+        require(values.size.isEven())
+
+        // Compute all twin sums and return the maximum
+        var maxTwinSum = 0
```

Comment:
Although the initial value 0 is valid for this problem (all values in the linkedList are positive), it is better to initialize it with an existing value such as the first twin sum.

Reply:
Resolved at https://github.com/hj-core/hj-leetcode-kotlin/pull/62/commits/28205c90bc38e55538032e60a0005ca71d27525e